src/libcore/macros.rs

```diff
@@ -108,6 +108,44 @@ macro_rules! assert_eq {
     });
 }
 
+/// Asserts that two expressions are not equal to each other.
+///
+/// On panic, this macro will print the values of the expressions with their
+/// debug representations.
+///
+/// # Examples
+///
+/// ```
+/// let a = 3;
+/// let b = 2;
+/// assert_ne!(a, b);
+/// ```
+#[macro_export]
+///#[stable(feature = "????", since = "????")]
+macro_rules! assert_eq {
+    ($left:expr , $right:expr) => ({
+        match (&$left, &$right) {
+            (left_val, right_val) => {
+                if *left_val == *right_val {
+                    panic!("assertion failed: `(left != right)` \
```

Comment:
i see you point, though i don't think you are right. i think it should be read something like: `assertion failed: (<insert the assertion that tried and failed here>)`.

for example:
with `assert_eq` it would print: `assertion failed: (6 == 7)`
https://play.rust-lang.org/?gist=d2cee1db97caf48c7ad7efc4c6cdc88f&version=stable&backtrace=0

as it is currently written, the `assert_ne` message is in keeping with the `assert_eq` behavior.